analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/DeliveringDebugFeaturesInProductionBase.cs

```diff
@@ -40,17 +41,19 @@ public abstract class DeliveringDebugFeaturesInProductionBase<TSyntaxKind> : Tra
 
         protected abstract TrackerBase<TSyntaxKind, InvocationContext>.Condition IsInvokedConditionally();
 
+        protected abstract TrackerBase<TSyntaxKind, InvocationContext>.Condition IsInDevelopmentContext();
```

Comment:
These two methods have very complicated signature that's propagated to inheriting classes. It's always puzzling to understand the lambda indirection here. I suggest update both of them like this:
```
protected abstract bool IsInDevelopmentContext(SyntaxNode node);
```
and use them like this:
```
t.ExceptWhen(c => IsInvokedConditionally(c.Node))
```

Reply:
Soo much easier to read :)